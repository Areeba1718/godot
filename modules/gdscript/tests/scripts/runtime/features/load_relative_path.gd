func test():
    var file1 = load("./load_relative_path_external.notest.gd")
    var inst1 = file1.new()
    inst1.hello()

    var file2 = load("./some_dir/../load_relative_path_external.notest.gd")
    var inst2 = file2.new()
    inst2.hello()

    var file3 = load("../features/load_relative_path_external.notest.gd")
    var inst3 = file3.new()
    inst3.hello()

    var withdot = load("./.load_relative_path_dot.notest.gd")
    var instwithdot = withdot.new()
    instwithdot.hello()

    var withdot2 = load("./..load_relative_path_dot.notest.gd")
    var instwithdot2 = withdot2.new()
    instwithdot2.hello()
