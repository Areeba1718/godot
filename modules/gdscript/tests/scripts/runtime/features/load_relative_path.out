GDTEST_OK
>> WARNING
>> Line: 3
>> UNSAFE_METHOD_ACCESS
>> The method "new()" is not present on the inferred type "Resource" (but may be present on a subtype).
>> WARNING
>> Line: 4
>> UNSAFE_METHOD_ACCESS
>> The method "hello()" is not present on the inferred type "Variant" (but may be present on a subtype).
>> WARNING
>> Line: 7
>> UNSAFE_METHOD_ACCESS
>> The method "new()" is not present on the inferred type "Resource" (but may be present on a subtype).
>> WARNING
>> Line: 8
>> UNSAFE_METHOD_ACCESS
>> The method "hello()" is not present on the inferred type "Variant" (but may be present on a subtype).
>> WARNING
>> Line: 11
>> UNSAFE_METHOD_ACCESS
>> The method "new()" is not present on the inferred type "Resource" (but may be present on a subtype).
>> WARNING
>> Line: 12
>> UNSAFE_METHOD_ACCESS
>> The method "hello()" is not present on the inferred type "Variant" (but may be present on a subtype).
>> WARNING
>> Line: 15
>> UNSAFE_METHOD_ACCESS
>> The method "new()" is not present on the inferred type "Resource" (but may be present on a subtype).
>> WARNING
>> Line: 16
>> UNSAFE_METHOD_ACCESS
>> The method "hello()" is not present on the inferred type "Variant" (but may be present on a subtype).
>> WARNING
>> Line: 19
>> UNSAFE_METHOD_ACCESS
>> The method "new()" is not present on the inferred type "Resource" (but may be present on a subtype).
>> WARNING
>> Line: 20
>> UNSAFE_METHOD_ACCESS
>> The method "hello()" is not present on the inferred type "Variant" (but may be present on a subtype).
loaded
loaded
loaded
withdot
withdot2
